### PR TITLE
Use builderWithExpectedSize when size is known

### PR DESCRIPTION
**Pull request by wendigo**
## Description

> Is this change a fix, improvement, new feature, refactoring, or other?

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

> How would you describe this change to a non-technical end user or system administrator?

## Documentation

( ) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

( ) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by hashhar** [comment]
Most of this is in engine, so cc: @sopel39 .

It looks good to me however.

File: core/trino-main/src/test/java/io/trino/RowPageBuilder.java [hunk 1]
```diff
@@ -45,7 +45,7 @@ public static RowPageBuilder rowPageBuilder(Iterable<Type> types)
     RowPageBuilder(Iterable<Type> types)
     {
         this.types = ImmutableList.copyOf(requireNonNull(types, "types is null"));
-        ImmutableList.Builder<BlockBuilder> builders = ImmutableList.builder();
+        ImmutableList.Builder<BlockBuilder> builders = ImmutableList.builderWithExpectedSize(this.types.size());
```
Comment (sopel39) [inline core/trino-main/src/test/java/io/trino/RowPageBuilder.java:48]: `this` is redundant

**Review by sopel39 — APPROVED** [review]
I'm not sure it matters for perf though

**Comment by sopel39** [comment]
mind automation

**Comment by wendigo** [comment]
@sopel39 according to javadoc builderWithExpectedSize is faster

**Comment by wendigo** [comment]
CI hit: https://github.com/trinodb/trino/issues/12413

**Comment by sopel39** [comment]
> @sopel39 according to javadoc builderWithExpectedSize is faster

it is, but places where it's used might be irrelevant.

**Comment by sopel39** [comment]
anyway, I think it's fine to land

**Comment by sopel39** [comment]
@wendigo ping me when tests pass

**Comment by wendigo** [comment]
@sopel39 ping

**Pull request merged**